lib/src/date_only.dart

```diff
@@ -0,0 +1,66 @@
+part of '../kiota_abstractions.dart';
+
+/// Interface for a date only object.
+///
+/// This interface provides an abstraction layer over date only objects.
+/// It is used to represent date only values in a serialization format agnostic
+/// way.
+///
+/// It can only be used to represent a date in the Gregorian calendar.
+abstract class DateOnly {
+  /// Extracts the date part of a [DateTime] and creates a new [_DateOnlyImpl].
+  factory DateOnly.fromDateTime(DateTime dateTime) {
```

Comment:
I think the following methods might be missing:
- to string which would serialize to the rfc 3339 representation
- from string, same but the other way around